src/store/hooks.ts

```diff
@@ -0,0 +1,5 @@
+import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
+import type { RootState, AppDispatch } from './store'
+
+export const useAppDispatch = () => useDispatch<AppDispatch>()
```

Comment:
Can you explain why we need to create separate file and in fact reimport `useDispatch` and `useSelector`?

Reply:
this trick was provided by redux toolkit documentation for comfortable using types
https://redux-toolkit.js.org/tutorials/typescript#define-typed-hooks